### PR TITLE
Upgrade to Gradle 6 with minimum required versions of plugins

### PR DESCRIPTION
**Pull request by chali**

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -17,20 +17,15 @@ buildscript {
         classpath "com.netflix.nebula:gradle-aggregate-javadocs-plugin:3.0.1"
         classpath "com.google.protobuf:protobuf-gradle-plugin:0.8.8"
         classpath "io.franzbecker:gradle-lombok:3.2.0"
-        /*
-         * TODO: Upgrading to 1.3.1 caused problems
-         * A problem was found with the configuration of task ':genie-ui:npmInstall'.
-         * > File '{somePath}/genie/genie-ui/package-lock.json' specified for property '$2' does not exist.
-         */
-        classpath "com.moowork.gradle:gradle-node-plugin:1.2.0"
+        classpath "com.netflix.nebula:nebula-node-plugin:1.3.0"
```
Comment (chali) [inline build.gradle]: This is a fork of original `gradle-node-plugin` which is not maintained anymore and it is not compatible with Gradle 6.

File: codequality/checkstyle/checkstyle.xml [hunk 2]
```diff
@@ -164,12 +164,6 @@
         <!-- Checks the placement of left curly braces on types, methods and other blocks. -->
         <module name="LeftCurly"/>
 
-        <!-- Checks for long lines. -->
```
Comment (chali) [inline codequality/checkstyle/checkstyle.xml]: This change is required by breaking change in checkstyle which is upgraded in Gradle.

File: build.gradle (hunk 1 above)
Comment (tgianos) [reply build.gradle]: Will this be maintained or should we be looking into some other options?

File: build.gradle [hunk 3]
```diff
@@ -17,20 +17,15 @@ buildscript {
...
         /*
          * TODO: Failed when upgrading to 2.0.0. Looks like a Guava version problem (surprise)
          * Execution failed for task ':genie-agent:spotbugsIntegTest'.
          * > Failed to run Gradle SpotBugs Worker
          *   > Could not create an instance of type com.github.spotbugs.internal.spotbugs.SpotBugsExecutor.
          *      > com.google.common.collect.ImmutableList.builderWithExpectedSize(I)Lcom/google/common/collect/ImmutableList$Builder;
          */
-        classpath "gradle.plugin.com.github.spotbugs:spotbugs-gradle-plugin:1.6.9"
+        classpath "com.github.spotbugs:spotbugs-gradle-plugin:3.0.0"
```
Comment (tgianos) [inline build.gradle]: If this fixes the issue we can probably get rid of the big comment block above this.

File: gradle/wrapper/gradle-wrapper.properties [hunk 4]
```diff
@@ -1,5 +1,5 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-5.6.3-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-6.0-bin.zip
```
Comment (tgianos) [inline gradle/wrapper/gradle-wrapper.properties]: the latest appears to be `6.0.1`. should we just go ahead and do that as part of this?

Comment (chali) [reply gradle/wrapper/gradle-wrapper.properties]: sounds good. Let me update that.

File: build.gradle (hunk 1 above)
Comment (chali) [reply build.gradle]: Yes, we plan to maintain compatibility. We don't plan any feature development.

**Comment by chali** [comment]
@tgianos `npm-shrinkwrap.json` has now its original content.

**Pull request merged**